proto/substrait/algebra.proto

```diff
@@ -52,6 +53,22 @@ message RelCommon {
       substrait.extensions.AdvancedExtension advanced_extension = 10;
     }
   }
+
+  // Optional metadata that the producer may attach to the relation.
+  // Consumer should hold no expectations about the existance metadata.
+  message Metadata {
+    // Name (alias) for this relation.
+    // If provided, must be unique across the relations in the plan/root.
+    // Safe to ignore, or can be used for e.g. qualifying the relation (see e.g.
+    // Spark's SubqueryAlias), or debugging.
+    // Repeated to allow for multiple levels of naming, e.g. catalog/schema/table.
+    repeated string alias = 1;
+
+    // Any custom metadata that the producer wants to attach to the relation.
+    // Consumer should hold no expectations about the existence, or lack of,
+    // or meaning of, of any specific key, unless the producer is known.
+    map<string, string> custom = 2;
```

Comment:
Having string values is probably the right choice here as it means you can't stuff arbitrary protobufs here (that's the job on a protobuf extension anyway).